File: 2172.java
import java.util.Scanner;
public class Main {
 static final double EPS =1e-7;

  //矩形x*y 能否放入 a*b 中
static  boolean canfit( double a, double b, double x, double y )
{
   double t;
   double l=Math.sqrt(x*x+y*y),ll=x*x+y*y;
   if(x*x+y*y - a*a-b*b > EPS)
    return false;

   if(x< y)
    {
    t=x;x=y;y=t;
    }
   if(a< b)
   {
    t=a;a=b;b=t;
   }
   if( x-b >EPS && y -b > EPS )return false;
    double s=(4*x*y*b+Math.sqrt(16*x*x*y*y*b*b-4*ll*(4*x*x*y*y-ll*ll+b*b*ll) ) )/ll/2;

   if(s-a > EPS)
    return false;
   else return true;
}

  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);


    double a, b, c, d, e;
    a=sc.nextDouble();
    b=sc.nextDouble();
    c=sc.nextDouble();
    d=sc.nextDouble();
    e=sc.nextDouble();

    if( canfit( d, e, a, b ) || canfit( d, e, a, c ) || canfit( d, e, b, c ) )
    System.out.printf( "YES\n" );
    else
    System.out.printf( "NO\n" );

  }
}

File: 2172.cc
<pre class = "lang:c++ decode:true">
#include &lt; math.h & gt;
#include &lt; stdio.h & gt;
#include &lt; stdlib.h & gt;

    struct point {
  double x, y, h;
} a, b;

void Swap(double &amp; a, double &amp; b) {
  double t;
  t = a;
  a = b;
  b = t;
}

bool check(double c, double d, double a, double b) {
  if (a * b & lt; c * d) return false;
  if (a & lt; b) Swap(a, b);
  if (c & lt; d) Swap(c, d);
  if (b & lt; d) return false;
  if (a & gt; = c & amp; &amp; b & gt; = d) return true;
  double dis = sqrt(c * c + d * d);
  double p = asin(b / dis);
  double q = asin(d / dis);
  double sita = p - q;
  double Len = c * cos(sita) + d * sin(sita);
  if (Len & lt; = a) return true;
  return false;
}

int main() {
  scanf("%lf %lf %lf %lf %lf", &amp; a.x, &amp; a.y, &amp; a.h, &amp; b.x, &amp;
        b.y);
  if (check(a.x, a.y, b.x, b.y) || check(a.x, a.h, b.x, b.y) ||
      check(a.y, a.h, b.x, b.y))
    puts("YES");
  else
    puts("NO");
  return 0;
}
